salt/modules/influxdbmod.py

```diff
@@ -434,6 +435,18 @@ def create_retention_policy(database,
     default : False
         Whether or not the policy as default will be set as default.
 
+    shard_duration : 0s
+        New shard duration of given retention policy.
+
+        Durations such as 1h, 90m, 12h, 7d, and 4w, are all supported
+        and mean 1 hour, 90 minutes, 12 hours, 7 day, and 4 weeks,
+        respectively. Infinite retention is not supported.
+        As a workaround, specify a “1000w” duration to achieve an extremely
+        long shard group duration.
+        Defaults to “0s”, which is interpreted by the database to mean the
+        default value given the duration.
+        The minimum shard group duration is 1 hour.
+
```

Comment:
will need to add `.. versionadded:: Neon` for each new argument added in this pr.